lib/shopify_api/clients/http_client.rb

```diff
@@ -32,8 +32,8 @@ def initialize(base_path:, session: nil)
         end
       end
 
-      sig { params(request: HttpRequest).returns(HttpResponse) }
-      def request(request)
+      sig { params(request: HttpRequest, graphql_response_object: T::Boolean).returns(HttpResponse) }
+      def request(request, graphql_response_object: false)
```

Comment:
This is an even safer approach, I like it!

Maybe we rename this param as `response_as_struct: false` to be a little more explicit as what the option is doing.